src/sentry/releasehealth/metrics.py

```diff
@@ -150,3 +158,185 @@ def _compute_crash_free_rate(data: Dict[str, float]) -> Optional[float]:
+                Condition(Column("timestamp"), Op.GTE, start),
+                Condition(Column("timestamp"), Op.LT, now),
+                Condition(
+                    Column(tag_key(org_id, "session.status")), Op.EQ, tag_value(org_id, "init")
+                ),
+            ]
+
+            if environments is not None:
+                environment_tag_values = []
+
+                for environment in environments:
+                    value = indexer.resolve(org_id, UseCase.TAG_VALUE, environment)  # type: ignore
+                    if value is not None:
+                        environment_tag_values.append(value)
+
+                where_common.append(
+                    Condition(Column(tag_key(org_id, "environment")), Op.IN, environment_tag_values)
+                )
+
+            if not total:
+                release_tag_values = []
+
+                for _, release in project_releases:
+                    value = indexer.resolve(org_id, UseCase.TAG_VALUE, release)  # type: ignore
+                    if value is not None:
+                        # We should not append the value if it hasn't been
+                        # observed before.
+                        release_tag_values.append(value)
+
+                where_common.append(
+                    Condition(Column(tag_key(org_id, "release")), Op.IN, release_tag_values)
+                )
+
+            return where_common
+
+        def _get_common_groupby(total: bool) -> List[SelectableExpression]:
+            if total:
+                return [Column("project_id")]
+            else:
+                return [Column("project_id"), Column(tag_key(org_id, "release"))]
+
+        def _convert_results(data: Any, total: bool) -> Dict[Any, int]:
+            if total:
+                return {x["project_id"]: x["value"] for x in data}
+            else:
+                release_tag = tag_key(org_id, "release")
+                return {(x["project_id"], x[release_tag]): x["value"] for x in data}
+
+        def _count_sessions(total: bool, referrer: str) -> Dict[Any, int]:
+            query = Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity("metrics_counters"),
+                select=[Column("value")],
+                where=_get_common_where(total)
+                + [
+                    Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session")),
+                ],
+                groupby=_get_common_groupby(total),
+            )
+
+            return _convert_results(
+                raw_snql_query(
+                    query,
+                    referrer=referrer,
+                    use_cache=False,
+                )["data"],
+                total=total,
+            )
+
+        def _count_users(total: bool, referrer: str) -> Dict[Any, int]:
+            query = Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity("metrics_sets"),
```

Comment:
Enum here would be better as well `EntityKey.METRICS_SET.value`

Reply:
True, will keep in mind for future PRs